runtime/src/accounts_background_service.rs

```diff
@@ -49,15 +49,80 @@ pub type SnapshotRequestReceiver = Receiver<SnapshotRequest>;
 pub type DroppedSlotsSender = Sender<(Slot, BankId)>;
 pub type DroppedSlotsReceiver = Receiver<(Slot, BankId)>;
 
+/// interval to report bank_drop queue events: 60s
+const BANK_DROP_SIGNAL_CHANNEL_REPORT_INTERVAL: u64 = 60_000;
+
+/// Bank drop signal queue events
+enum BankDropQueueEvent {
+    Full,
+    Disconnected,
+}
+
+/// Bank drop signal queue event statistics
+#[derive(Debug, Default)]
+struct BankDropQueueStats {
+    report_time: AtomicU64,
+    queue_full: AtomicUsize,
+    queue_disconnected: AtomicUsize,
+}
+
+impl BankDropQueueStats {
+    /// submit bank drop signal queue event counters
+    fn report(&self, event: BankDropQueueEvent) {
+        let counter = match event {
+            BankDropQueueEvent::Full => &self.queue_full,
+            BankDropQueueEvent::Disconnected => &self.queue_disconnected,
+        };
+
+        let name = match event {
+            BankDropQueueEvent::Full => "full",
+            BankDropQueueEvent::Disconnected => "disconnected",
+        };
+
+        let ts = solana_sdk::timing::timestamp();
+
+        counter.fetch_add(1, Ordering::Relaxed);
+        let last_report_time = self.report_time.load(Ordering::Acquire);
+        if ts.saturating_sub(last_report_time) > BANK_DROP_SIGNAL_CHANNEL_REPORT_INTERVAL {
+            let val = counter.load(Ordering::Relaxed);
+
+            if counter
+                .compare_exchange_weak(val, 0, Ordering::AcqRel, Ordering::Acquire)
+                .is_ok()
+            {
+                if val > 0 {
+                    datapoint_info!("bank_drop_queue_event", (name, val, i64));
+                }
+                self.report_time.store(ts, Ordering::Release);
+            }
+        }
+    }
+}
+
+lazy_static! {
+    static ref BANK_DROP_QUEUE_STATS: BankDropQueueStats = BankDropQueueStats::default();
+}
+
 #[derive(Clone)]
 pub struct SendDroppedBankCallback {
     sender: DroppedSlotsSender,
 }
 
 impl DropCallback for SendDroppedBankCallback {
     fn callback(&self, bank: &Bank) {
-        if let Err(e) = self.sender.send((bank.slot(), bank.bank_id())) {
-            warn!("Error sending dropped banks: {:?}", e);
+        match self.sender.try_send((bank.slot(), bank.bank_id())) {
+            Err(TrySendError::Full(_)) => {
+                BANK_DROP_QUEUE_STATS.report(BankDropQueueEvent::Full);
```

Comment:
If we get a burst of these events in 60 seconds, but then none after, so we never hit the `if ts.saturating_sub(last_report_time) > BANK_DROP_SIGNAL_CHANNEL_REPORT_INTERVAL {` we may not report.

To deal with this we could call `report()` on every `BANK_DROP_SIGNAL_CHANNEL_REPORT_INTERVAL` and see if there's anything to report, but maybe this is small enough of edge case where it's not worth it

Reply:
good observation. sure. will fix.